apps/api/src/scraper/scrapeURL/transformers/index.ts

```diff
@@ -31,6 +31,35 @@ export function deriveMetadataFromRawHTML(
   return document;
 }
 
+export function encodeRawHTML(_meta: Meta, document: Document): Document {
+  if (document.rawHtml === undefined) {
+    throw new Error(
+      "rawHtml is undefined -- this transformer is being called out of order",
+    );
+  }
+
+  const data: Uint8Array = new TextEncoder().encode(document.rawHtml);
+
+  const charsetMatch = document.rawHtml.match(
+    /<meta[^>]+charset=["']?([^"'>]+)/i,
+  );
+  const charset = charsetMatch ? charsetMatch[1] : "UTF-8";
+
+  try {
+    // Convert the response data if charset is not UTF-8
+    if (charset.toUpperCase() !== "UTF-8") {
+      const decoder = new TextDecoder(charset);
+      document.rawHtml = decoder.decode(data);
+      
+      return document;
+    }
```

Comment:
The approach is good, but I think we need to move it to earlier parts of the stack, since the UTF-8 misencode is too destructive to get usable Shift_JIS out of just the parsed string. Let's talk about this on Monday.